pkg/ingestor/parser/cyclonedx/parser_cyclonedx.go

```diff
@@ -95,24 +95,31 @@ func (c *cyclonedxParser) CreateEdges(ctx context.Context, foundIdentities []ass
 func (c *cyclonedxParser) addRootPackage(cdxBom *cdx.BOM) {
 	// oci purl: pkg:oci/debian@sha256%3A244fd47e07d10?repository_url=ghcr.io/debian&tag=bullseye
 	if cdxBom.Metadata.Component != nil {
-		splitImage := strings.Split(cdxBom.Metadata.Component.Name, "/")
-		if len(splitImage) == 3 {
-			rootPackage := assembler.PackageNode{}
-			rootPackage.Purl = "pkg:oci/" + splitImage[2] + "?repository_url=" + splitImage[0] + "/" + splitImage[1]
-			rootPackage.Name = cdxBom.Metadata.Component.Name
+		rootPackage := assembler.PackageNode{}
+		rootPackage.Name = cdxBom.Metadata.Component.Name
+		rootPackage.NodeData = *assembler.NewObjectMetadata(c.doc.SourceInformation)
+		if cdxBom.Metadata.Component.PackageURL != "" {
+			rootPackage.Purl = cdxBom.Metadata.Component.PackageURL
 			rootPackage.Version = cdxBom.Metadata.Component.Version
-			rootPackage.Digest = append(rootPackage.Digest, cdxBom.Metadata.Component.Version)
-			rootPackage.Tags = []string{"CONTAINER"}
-			rootPackage.NodeData = *assembler.NewObjectMetadata(c.doc.SourceInformation)
-			c.rootPackage = parentPackages{
-				curPackage:  rootPackage,
-				depPackages: []parentPackages{},
+			rootPackage.Tags = []string{string(cdxBom.Metadata.Component.Type)}
+		} else {
+			splitImage := strings.Split(cdxBom.Metadata.Component.Name, "/")
+			if len(splitImage) == 3 {
+				rootPackage.Purl = "pkg:oci/" + splitImage[2] + "?repository_url=" + splitImage[0] + "/" + splitImage[1]
+				rootPackage.Version = cdxBom.Metadata.Component.Version
+				rootPackage.Digest = append(rootPackage.Digest, cdxBom.Metadata.Component.Version)
+				rootPackage.Tags = []string{"CONTAINER"}
 			}
 		}
+		c.rootPackage = parentPackages{
+			curPackage:  rootPackage,
+			depPackages: []*parentPackages{},
+		}
 	}
 }
 
 func (c *cyclonedxParser) addPackages(cdxBom *cdx.BOM) {
+	pkgMap := map[string]*parentPackages{}
```

Comment:
i think it may be helpful to have this pkgMap be in the parser struct since the references will likely will be useful in other aspects in the future.

Reply:
done!